### PR TITLE
Add note about peephole optimizer in docs

### PR DESCRIPTION
**Pull request by matheusaaguiar**

### CODE REVIEW

File: docs/internals/optimizer.rst [hunk 1]
```diff
@@ -26,6 +26,10 @@ One can use ``solc --ir-optimized --optimize`` to produce an
 optimized Yul IR for a Solidity source. Similarly, one can use ``solc --strict-assembly --optimize``
 for a stand-alone Yul mode.
 
+.. note::
+    The PeepholeOptimizer is always enabled by default and can only be turned off via the
```
Comment (nikola-matic) [inline docs/internals/optimizer.rst]: Would it make sense to mention what the peephole optimizer is, since I don't think we have it documented anywhere?

Comment (matheusaaguiar) [reply docs/internals/optimizer.rst]: From what I gather, `Peephole` refers to a common optimization method, so I guess that's why it does not have its own doc section. Maybe we could add an external link ([wikipedia article](https://en.wikipedia.org/wiki/Peephole_optimization)) as a reference to the technique itself.

Comment (matheusaaguiar) [reply docs/internals/optimizer.rst]:
```suggestion
    The `PeepholeOptimizer <https://en.wikipedia.org/wiki/Peephole_optimization>`_ is always enabled by default and can only be turned off via the
```

**Comment by nikola-matic** [comment]
Pull `develop` and rebase against it to get the build to a merge-able state pls :)

**Comment by nikola-matic** [comment]
By the way, did you check if the linking to the JSON input output works properly with `docs/docs.sh`? Otherwise, LGTM.

**Review by nikola-matic — DISMISSED** [review]

**Comment by matheusaaguiar** [comment]
> By the way, did you check if the linking to the JSON input output works properly with `docs/docs.sh`? Otherwise, LGTM.

Yeah, checked it.
Rebased on develop now.

**Review by nikola-matic — DISMISSED** [review]

**Comment by nikola-matic** [comment]
> > By the way, did you check if the linking to the JSON input output works properly with `docs/docs.sh`? Otherwise, LGTM.
> 
> Yeah, checked it. Rebased on develop now.

Approved again, although you might wanna wait for @ekpyron, @chriseth or @cameel as per the new policy?

**Comment by matheusaaguiar** [comment]
I guess it is not really needed, in this case. Although, I do prefer the merge is done by someone else rather than PR author.
Anyway, this is not urgent :)

File: docs/internals/optimizer.rst [hunk 2]
```diff
@@ -26,6 +26,10 @@ One can use ``solc --ir-optimized --optimize`` to produce an
 optimized Yul IR for a Solidity source. Similarly, one can use ``solc --strict-assembly --optimize``
 for a stand-alone Yul mode.
 
+.. note::
+    The `PeepholeOptimizer <https://en.wikipedia.org/wiki/Peephole_optimization>`_ is always
+    enabled by default and can only be turned off via the :ref:`JSON-input-output interface <compiler-api>`.
```
Comment (cameel) [inline docs/internals/optimizer.rst]: Peephole optimizer is not the only such step. The other one is the inliner:
>        // The inliner is always on if no details are given,
>        // use details to switch it off.
>        "inliner": true,

Can you mention it too?

Comment (matheusaaguiar) [reply docs/internals/optimizer.rst]: Done.

File: docs/internals/optimizer.rst [hunk 3]
```diff
@@ -26,6 +26,10 @@ One can use ``solc --ir-optimized --optimize`` to produce an
 optimized Yul IR for a Solidity source. Similarly, one can use ``solc --strict-assembly --optimize``
 for a stand-alone Yul mode.
 
+.. note::
+    The `PeepholeOptimizer <https://en.wikipedia.org/wiki/Peephole_optimization>`_ and the inliner are always
```
Comment (cameel) [inline docs/internals/optimizer.rst]: Are we referring to optimizer steps or just using generic names? I'd make that consistent one way or another.
```suggestion
    The `peephole optimizer <https://en.wikipedia.org/wiki/Peephole_optimization>`_ and the inliner are always
```

File: docs/internals/optimizer.rst [hunk 4]
```diff
@@ -26,6 +26,10 @@ One can use ``solc --ir-optimized --optimize`` to produce an
...
+    enabled by default and can only be turned off via the :ref:`JSON-input-output interface <compiler-api>`.
```
Comment (cameel) [inline docs/internals/optimizer.rst]: We usually refer to it as Standard JSON:
```suggestion
    enabled by default and can only be turned off via the :ref:`Standard JSON <compiler-api>`.
```

**Review by cameel — DISMISSED** [review]
I noticed two tiny details that might be good to change but overall the PR is good.

File: docs/internals/optimizer.rst (hunk 3 above)
Comment (matheusaaguiar) [reply docs/internals/optimizer.rst]: The other occurrences of the word are as you suggested, so I am changing it.

File: docs/internals/optimizer.rst (hunk 4 above)
Comment (matheusaaguiar) [reply docs/internals/optimizer.rst]: Ok. Done.

**Review by cameel — APPROVED** [review]

**Pull request merged**